### PR TITLE
feat: Add KMS Keys to s3 Destination 

### PR DESCRIPTION
**Pull request by sunil494**
#### Summary

⚠️ **If you're contributing to a plugin please read this section of the [contribution guidelines](https://github.com/cloudquery/cloudquery/blob/main/CONTRIBUTING.md#open-core-vs-open-source) 🧑‍🎓 before submitting this PR** ⚠️

### CODE REVIEW

File: plugins/destination/s3/client/spec/spec.go [hunk 1]
```diff
@@ -61,6 +63,12 @@ type Spec struct {
 	// **Note**: if you want to use path-style addressing, i.e., `https://s3.amazonaws.com/BUCKET/KEY`, `use_path_style` should be enabled, too.
 	Endpoint string `json:"endpoint,omitempty"  jsonschema:"default="`
 
+	// SSE KMS Key ID appened to S3 API calls header. Used in conjuction with server_side_encryption.
+	SSEKMSKeyId string `json:"sse_kms_key_id,omitempty" jsonschema:"default="`
+
+	//Server Side Encryption header which declares encryption type in S3 API calls header: x-amz-server-side-encryption.
+	ServerSideEncryption types.ServerSideEncryption `json:"server_side_encryption,omitempty" jsonschema:"default="`
```
Comment (disq) [inline plugins/destination/s3/client/spec/spec.go:70]:
```suggestion
	ServerSideEncryption types.ServerSideEncryption `json:"server_side_encryption,omitempty" jsonschema:"enum=AES256,enum=aws:kms,enum=aws:kms:dsse"`
```

Comment (mnorbury) [reply plugins/destination/s3/client/spec/spec.go:70]: Also need to run `make gen-spec-schema` to update the JSON schema

Comment (mnorbury) [reply plugins/destination/s3/client/spec/spec.go]: I believe you are correct, the `read.go` shouldn't need changing.

**Review by mnorbury — APPROVED** [review]
This looks good to me. Thank you for your contribution.

I have some changes I would like to make to the JSON schema validation, but I can do this in a follow-up PR.

**Review by disq — APPROVED** [review]

**Comment by mnorbury** [comment]
Hi @sunil494 , thanks again for your contribution. This is now merged and should be available in [v4.10.0](https://github.com/cloudquery/cloudquery/pull/16786) of the plugin.

ℹ️ To make the schema validation easier, the server encryption configuration parameters are configured in a nested block e.g. see below:

```yaml
kind: destination
spec:
  name: "s3"
  path: "cloudquery/s3"
  registry: "cloudquery"
  version: "v4.10.0"
  write_mode: "append"
  spec:
    bucket: "<bucket name>"
    region: "eu-west-1" # Example: us-east-1
    path: "path/to/files/{{TABLE}}/data.json"
    format: "json" # options: parquet, json, csv
    no_rotate: true
    server_side_encryption_configuration:
       sse_kms_key_id: "<key id>"
       server_side_encryption: "aws:kms"
```

**Pull request closed without merging**